apply/apply.go

```diff
@@ -32,49 +33,54 @@ func Apply(fs afero.Fs, conf *config.Config, tmp *templates.T, upgrade bool) err
 		toolVersion, _ := util.VersionString()
 		versionChange, repoVersion, _ := checkToolVersions(fs, toolVersion)
 		if versionChange {
-			return fmt.Errorf("fogg version (%s) is different than version currently used to manage repo (%s). To upgrade add --upgrade.", toolVersion, repoVersion)
+			return errs.NewUserf("fogg version (%s) is different than version currently used to manage repo (%s). To upgrade add --upgrade.", toolVersion, repoVersion)
 		}
 	}
 	p, err := plan.Eval(conf, false)
 	if err != nil {
-		return errors.Wrap(err, "unable to evaluate plan")
+		return errs.WrapUser(err, "unable to evaluate plan")
 	}
 
 	e := applyRepo(fs, p, &tmp.Repo)
 	if e != nil {
-		return errors.Wrap(e, "unable to apply repo")
+		return errs.WrapUser(e, "unable to apply repo")
+	}
+
+	e = applyPlugins(fs, p)
+	if e != nil {
+		return errs.WrapUser(e, "unable to apply plugins")
 	}
 
 	e = applyAccounts(fs, p, &tmp.Account)
 	if e != nil {
-		return errors.Wrap(e, "unable to apply accounts")
+		return errs.WrapUser(e, "unable to apply accounts")
 	}
 
 	e = applyEnvs(fs, p, &tmp.Env, &tmp.Component)
 	if e != nil {
-		return errors.Wrap(e, "unable to apply envs")
+		return errs.WrapUser(e, "unable to apply envs")
 	}
 
 	e = applyGlobal(fs, p.Global, &tmp.Global)
 	if e != nil {
-		return errors.Wrap(e, "unable to apply global")
+		return errs.WrapUser(e, "unable to apply global")
 	}
 
 	e = applyModules(fs, p.Modules, &tmp.Module)
-	return errors.Wrap(e, "unable to apply modules")
+	return errors.Wrap(e, "unable to apply modules") // FIXME
```

Comment:
This and a few other FIXMEs are here because when I change them things break (not the tests, but when running against an example repo). @edulop91 it is all related to plugins, maybe we can pair on fixing tomorrow?